File: framework/languages/java/e2e/DiscoveryRegistryHa/Client/src/main/java/systems/zlink/e2e/discoveryregistryha/client/support/ScenarioAssert.java
package systems.zlink.e2e.discoveryregistryha.client.support;

public final class ScenarioAssert {
    private ScenarioAssert() {
    }

    public static void that(boolean condition, String message) {
        if (!condition) {
            throw new IllegalStateException(message);
        }
    }
}

File: framework/languages/java/e2e-kotlin/RegistryMessaging/Client/src/main/kotlin/systems/zlink/e2e/kotlin/registrymessaging/client/Support/ScenarioAssert.kt
package systems.zlink.e2e.kotlin.registrymessaging.client.Support

object ScenarioAssert {
    fun that(condition: Boolean, message: String) {
        if (!condition) {
            throw IllegalStateException(message)
        }
    }

    fun countNewEvidence(after: Collection<String>, before: Collection<String>, prefix: String, marker: String): Int {
        val beforeCount = before.count { it.contains(prefix) && it.contains(marker) }
        val afterCount = after.count { it.contains(prefix) && it.contains(marker) }
        return afterCount - beforeCount
    }
}
